internal/controllers/task.go

```diff
@@ -0,0 +1,122 @@
+package controllers
+
+import (
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+
+	"github.com/thatnerdjosh/example-webservices/internal/config"
+	"github.com/thatnerdjosh/example-webservices/internal/models"
+)
+
+type HttpClient interface {
+	Get(string) (*http.Response, error)
+}
+
+type TaskController struct {
+	config *config.TaskConfig
+	client HttpClient
+}
+
+func NewTaskController(taskConfig *config.TaskConfig, client HttpClient) TaskController {
+	var ctrl TaskController
+
+	ctrl.config = &config.TaskConfig{}
+	if taskConfig != nil {
+		ctrl.config = taskConfig
+	}
+
+	ctrl.client = client
+
+	ctrl.config.MustLoad()
+	return ctrl
+}
+
+func (t TaskController) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	switch {
+	case r.Method == "POST" && r.URL.Path == "/":
+		t.ExecuteTask(w, r)
+	default:
+		w.WriteHeader(http.StatusNotFound)
+	}
+}
+
+func (t TaskController) ExecuteTask(w http.ResponseWriter, r *http.Request) {
```

Comment:
I specifically was trying to avoid this,  but I might have missed some business logic in the controller - oh right --- I think you might be referring to the authentication check. Good point. I'll resolve this.